src/overlays/actors/ovl_Obj_Demo/z_obj_demo.c

```diff
@@ -29,14 +29,90 @@ const ActorInit Obj_Demo_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjDemo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDemo* this = THIS;
+    s16 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/ObjDemo_Init.s")
+    thisx->params &= 0xFF;
+    if ((thisx->params != 0xFF) && (Flags_GetSwitch(globalCtx, thisx->params))) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        if (thisx->shape.rot.x < 0) {
+            this->unk_148 = 1;
+            thisx->shape.rot.x = -thisx->shape.rot.x;
+        }
+        if (thisx->shape.rot.x == 0) {
+            this->unk_14C = 80.0f;
+        } else {
+            this->unk_14C = thisx->shape.rot.x * 4.0f;
+            thisx->shape.rot.x = 0;
+            thisx->world.rot.x = 0;
+        }
+        temp_v0 = thisx->shape.rot.z;
+        if (temp_v0 == 0) {
+            this->unk_150 = 60.0f;
+        } else {
+            thisx->shape.rot.z = 0;
+            thisx->world.rot.z = 0;
+            this->unk_150 = temp_v0 * 4.0f;
+        }
+        func_80983634(globalCtx);
+        this->actionFunc = func_80983678;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/func_80983634.s")
+void func_80983634(GlobalContext* globalCtx) {
+    if ((globalCtx->sceneNum == SCENE_CASTLE) && (func_801A8A50(0) == NA_BGM_IKANA_CASTLE)) {
+        Audio_QueueSeqCmd(0x100100FF);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/func_80983678.s")
+void func_80983678(ObjDemo* this, GlobalContext* globalCtx) {
+    func_80983634(globalCtx);
+    if ((this->actor.xzDistToPlayer < this->unk_14C) && (fabsf(this->actor.playerHeightRel) < this->unk_150)) {
+        if (this->unk_148 == 1) {
+            ActorCutscene_Stop(0x7D);
+        }
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        this->actionFunc = func_80983704;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/func_80983704.s")
+void func_80983704(ObjDemo* this, GlobalContext* globalCtx) {
+    if ((this->unk_148 == 1) && (ActorCutscene_GetCurrentIndex() == 0x7D)) {
+        ActorCutscene_Stop(0x7D);
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        return;
+    }
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
```

Comment:
```suggestion
    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
```

Reply:
Can also do an `else if` to remove the early `return`